trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/interpolantautomata/builders/AbsIntNonSmtInterpolantAutomatonBuilder.java

```diff
@@ -95,24 +96,19 @@ private NestedWordAutomaton<LETTER, IPredicate> getPathProgramAutomatonNew(
 			final INwaOutgoingLetterAndTransitionProvider<LETTER, IPredicate> oldAbstraction,
 			final IPredicateUnifier predicateUnifier, final IEmptyStackStateFactory<IPredicate> emptyStackFactory) {
 		mLogger.info("Creating interpolant automaton from AI using abstract post for generalization");
-
-		final NestedRun<LETTER, IPredicate> cex = (NestedRun<LETTER, IPredicate>) mCurrentCounterExample;
-		final List<IPredicate> stateSequence = cex.getStateSequence();
-
-		if (stateSequence.size() <= 1) {
+		if (mControlConfigurationSequence.size() <= 1) {
```

Comment:
Can this still happen?

Reply:
It is still possible to have a `Counterexample<L>` with the empty word, and exactly 1 state.

Of course, this class should never be used for such a counterexample, but to detect it if it is seems exactly the point of the check here.

Reply:
Hm, would it make sense to forbid this in `Counterexample` instead? Here, at least one should check if `cex` has a `getStateSequence``to avoid the NPE.

Reply:
We would not get an NPE: The call to `getControlConfigurations()` in the constructor of this class would `throw new IllegalStateException("Counterexample with control configurations is required, but this counterexample does not have control configurations.");`.

Nevertheless, it does seem reasonable to only allow non-empty `Counterexample`s. I will add such a check.